x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -509,6 +512,16 @@ private void prewarmCache(ActionListener<Void> listener) {
         }
     }
 
+    private boolean isSnapshotFileFullyCached(BlobStoreIndexShardSnapshot.FileInfo file) {
+        try {
+            final CacheKey cacheKey = createCacheKey(file.physicalName());
+            final CacheFile cacheFile = getCacheFile(cacheKey, file.length());
```

Comment:
There are no guarantee that the CacheFile returned here is the same as the one used during prewarming, maybe we should instead compute the recovered bytes directly during prewarming?